Orange/widgets/visualize/tests/test_owtreegraph.py

```diff
@@ -35,3 +43,43 @@ def test_target_class_changed(self):
         self.widget.color_combo.activated.emit(1)
         self.widget.color_combo.setCurrentIndex(1)
         self.assertNotEqual(nodes[0].toPlainText(), text)
+
+    def test_tree_determinism(self):
+        """Check that the tree is drawn identically upon receiving the same
+        dataset with no parameter changes."""
+        n_tries = 10
+
+        def _check_all_same(data):
+            """Check that all the elements within an iterable are identical."""
+            iterator = iter(data)
+            try:
+                first = next(iterator)
+            except StopIteration:
+                return True
+            return all(first == rest for rest in iterator)
```

Comment:
Unfortunately not, because I'm comparing `QPointF`s which are an unhashable type.